### PR TITLE
Alt 1205

### PR DESCRIPTION
**Pull request by Dantheman720**
@LightGuard Maybe this would be better for adding additional parameters in the future?  It keeps the if/else blocks from getting out of hand in the future.  I'm not sure if this works though, I haven't been able to get testing up and running.

### CODE REVIEW

File: lib/aweplug/books/google_books.rb [hunk 1]
```diff
@@ -48,15 +48,15 @@ def initialize site, push_to_searchisko
         @client = google_client(site, authenticate: site.authenticate_google_books_api)
         @books = @client.discovered_api(BOOKS_API_SERVICE_NAME, BOOKS_API_VERSION)
         @searchisko = Aweplug::Helpers::Searchisko.default site, 360
+        @env_param = {
+            :q => "isbn:#{data['isbn']}"}
       end
 
       def get data
+        @env_param.merge!(:country => ENV['country_code']) if ENV['country_code']
```
Comment (LightGuard) [inline lib/aweplug/books/google_books.rb]: Uppercase `country_code`

Comment (Dantheman720) [inline lib/aweplug/books/google_books.rb]: Will fix and resubmit.

~Dan

**Comment by LightGuard** [comment]
:+1:

**Comment by LightGuard** [comment]
@paulrobinson We need to get the CI environments setup before we can merge this.

File: lib/aweplug/books/google_books.rb [hunk 2]
```diff
@@ -85,7 +85,7 @@ def get data
           unless required_keys.all? {|key| book['volumeInfo'].key? key}
             isbn = isbn_13(book) || data['isbn']
             puts "book: #{isbn} missing required attributes: #{required_keys - book['volumeInfo'].keys}"
-            return nil
```
Comment (paulrobinson) [inline lib/aweplug/books/google_books.rb]: don't combine whitespace changes with code changes. It makes it hard to detect the actual changes.

File: lib/aweplug/books/google_books.rb [hunk 3]
```diff
@@ -96,7 +96,7 @@ def get data
               thumbnail = book['volumeInfo']['imageLinks']['thumbnail']
             else
               thumbnail = cdn("#{@site.base_url}/images/books/book_noimageavailable.jpg")
-            end
```
Comment (paulrobinson) [inline lib/aweplug/books/google_books.rb]: more ws changes.

File: lib/aweplug/books/google_books.rb [hunk 4]
```diff
@@ -136,8 +136,8 @@ def get data
 
       def send_to_searchisko book
         unless !@push_to_searchisko || @site.profile =~ /development/
-          @searchisko.push_content('jbossdeveloper_book',
```
Comment (paulrobinson) [inline lib/aweplug/books/google_books.rb]: more ws changes.

File: lib/aweplug/books/google_books.rb [hunk 5]
```diff
@@ -169,7 +169,7 @@ def isbn_13 book
             return ids['OTHER']
           end
         end
-        nil
```
Comment (paulrobinson) [inline lib/aweplug/books/google_books.rb]: more ws changes.

**Comment by Dantheman720** [comment]
This doesn't work.  I didn't see that 'data' is a method variable.

**Comment by LightGuard** [comment]
I'm confused, what part isn't working?

On Monday, February 16, 2015, Dantheman720 <notifications@github.com> wrote:

> This doesn't work. I didn't see that 'data' is a method variable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/awestruct/aweplug/pull/89#issuecomment-74595263>.
>


-- 
Jason Porter
http://en.gravatar.com/lightguardjp

**Comment by Dantheman720** [comment]
I changed the instantiation of env_param to be inside of def get data.  I was referencing the method variable data from out of scope.

Sidenote, is there a good way to ignore whitespace changes when commiting?  I tried git apply --ignore-space-change, but it just hangs...

**Comment by LightGuard** [comment]
Which one are we keeping?

**Comment by Dantheman720** [comment]
This can be closed.  Obsolete

**Pull request closed without merging**